src/EventListener/UserEventSubscriber.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\EventListener;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Security\Core\AuthenticationEvents;
+use Symfony\Component\Security\Core\Event\AuthenticationFailureEvent;
+use Symfony\Component\Security\Http\SecurityEvents;
+use Symfony\Component\Security\Http\Event\InteractiveLoginEvent;
+use Monolog\Logger;
+use Doctrine\ORM\EntityManagerInterface;
+use Symfony\Component\HttpFoundation\Session\SessionInterface;
+use App\Entity\User;
+use App\Util\MonologDoctrineHandler;
+
+class UserEventSubscriber implements EventSubscriberInterface
+{
+    private $logger;
+    private $mailer;
+    private $entityManager;
+    private $session;
+
+    public function __construct(
+        \Swift_Mailer $mailer,
+        EntityManagerInterface $entityManager,
+        SessionInterface $session,
+        MonologDoctrineHandler $handler
+    ) {
+        $this->logger = new Logger('User Events');
+        $this->logger->pushHandler($handler);
+        $this->mailer = $mailer;
+        $this->entityManager = $entityManager;
+        $this->session = $session;
+    }
+
+    public static function getSubscribedEvents()
+    {
+        return [
+            AuthenticationEvents::AUTHENTICATION_FAILURE => 'onAuthenticationFailure',
+            SecurityEvents::INTERACTIVE_LOGIN => 'onAuthenticationSuccess'
+        ];
+    }
+
+    public function onAuthenticationFailure(AuthenticationFailureEvent $event)
+    {
+        $lastEmail = $event->getAuthenticationToken()->getUsername();
+
+        $this->logger->warning('Authentication Failure', ['username' => $lastEmail]);
+
+        if ($this->entityManager->getRepository(User::class)->findOneBy(['email' => $lastEmail]) != null) {
```

Comment:
@JanMaria Zawsze stosuj **identyczne porównywanie** z wyjątkiem sytuacji kiedy musisz żonglować typami (czyli na Twoim poziomie to nigdy :)). Zaś w sytuacjach kiedy chcesz zrobić porówanie, a nie chcesz zmieniać typu zmiennej to zrzuć zmienną na potrzebny typ **w locie**. **Przykład:** Masz zmienną typu string, a potrzebujesz porównać ją z liczbą `1 == $jedenAleTypuString`. Wtedy robisz tak `1 === (int) $jedenAleTypuString`